File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        long num = Long.parseLong(p);
        int size = p.length();
        
        for(int i = 0; i <= t.length()-size; i++) 
            if(num >= Long.parseLong(t.substring(i, i+size))) 
                answer++;

        return answer;
    }
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    st = len(t)
    sp = len(p)
    
    pi = int(p)
    
    cnt=0
    for i in range(st-sp+1):
        ti = int(t[i:i+sp])
        if ti <= pi:
            cnt+=1
    return cnt
    
    
    
    
    return answer